### PR TITLE
Run tests directly on GA runner

### PR DESCRIPTION
**Pull request by Patrik-Stas**
This speeds up CI significantly 

- Run rust tests, nodejs test, npmjs publishing directly on GA runners instead of in docker
- Removed build of libvcx-tester image, codecov image anymore
- Removed publishing jobs `publish-docker-libvcx-tester`, `publish-docker-codecov`, `publish-docker-android` - though they are still published under `alpine-core:<branch-name>` as part of `build-image` composite action (can be useful to inspect them for debugging)
- Well and apparently delta of about minus 500 lines as cherry on top

Signed-off-by: Patrik Stas <patrik.stas@absa.africa>

### CODE REVIEW

**Review by mirgee — CHANGES_REQUESTED** [review]
Very nice, our GA are shaping up to be a thing of beauty, good job! 👍 

For some reason, it seems as if the pipeline was run twice:
image
image

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -367,243 +207,121 @@ jobs:
...
-                              sh -c '(RUST_TEST_THREADS=1 TEST_POOL_IP=127.0.0.1 cargo test \
-                                    -F "mysql_test" \
-                                    --manifest-path="$HOME/aries_vcx/Cargo.toml")'
-      - name: "Collect docker logs on failure"
-        if: failure()
-        uses: ./.github/actions/upload-docker-logs
-        with:
-          name: "docker-services-${{ github.job }}"
+          cd wrappers/node
+          sudo cp "../../target/debug/libvcx.so" /usr/lib
```
[the remark this answers is not in the document]
Comment (Patrik-Stas) [reply .github/workflows/main.yml:301]: touche, it was mistake

File: .github/actions/setup-testing-nodejs/action.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        node-version: 12.x
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: 1.64.0
+    - uses: Swatinem/rust-cache@v2
+    - name: "Install dependencies"
+      shell: bash
+      run: |
+        sudo apt-get update -y
+        sudo apt-get install -y libsodium-dev libssl-dev libzmq3-dev
```
[the remark this answers is not in the document]
Comment (Patrik-Stas) [reply .github/actions/setup-testing-nodejs/action.yml:25]: no, but this is typically quite fast; I am not sure what would be reasonable way to cache this, I assume apt-get installing packages might in general create various files in various directories

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -883,39 +494,29 @@ jobs:
...
       - build-and-publish-android-device
       - build-and-publish-android-emulator
-      - test-image-general_test
-      - test-libvcx-image-pool-tests
-      - test-aries-vcx-image-pool-tests
-      - test-libvcx-image-agency-pool-tests
+      - test-unit-workspace
+      - test-integration-libvcx
+      - test-integration-aries-vcx
+      - test-integration-aries-vcx-mysql
```
[the remark this answers is not in the document]
Comment (Patrik-Stas) [reply .github/workflows/main.yml:519]: done so (for now removing the ones I've removed, we can add the renames after this is merged)

File: .github/actions/setup-testing-nodejs/action.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mirgee) [reply .github/actions/setup-testing-nodejs/action.yml:25]: It seems to take ~ 1 min, which is acceptable :)

**Review by mirgee — APPROVED** [review]

**Pull request merged**